clippy_lints/src/print.rs

```diff
@@ -161,24 +169,15 @@ fn check_print<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, span: Span, args: &HirVec<E
 }
 
 /// Check for println!("")
-fn check_println<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, span: Span, args: &HirVec<Expr>) {
+fn check_println<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, span: Span, fmtstr: InternedString, fmtlen: usize) {
     if_let_chain!{[
-        // ensure we're calling Arguments::new_v1
-        args.len() == 1,
-        let ExprCall(ref args_fun, ref args_args) = args[0].node,
-        let ExprPath(ref qpath) = args_fun.node,
-        let Some(const_def_id) = opt_def_id(resolve_node(cx, qpath, args_fun.hir_id)),
-        match_def_path(cx.tcx, const_def_id, &paths::FMT_ARGUMENTS_NEWV1),
-        args_args.len() == 2,
-
         // check that the string is empty
-        let Some((fmtstr, 1)) = get_argument_fmtstr_parts(&args_args[0]),
+        fmtlen == 1,
         fmtstr.deref() == "\n",
 
-        // HACK: check size of the macro invocation:
-        // println!();  // 11 chars: don't lint
-        // println!("") // 12 chars: lint
-        (span.hi() - span.lo()).0 > 11,
+        // check the presence of that string
+        let Ok(snippet) = cx.sess().codemap().span_to_snippet(span),
+        snippet.contains("\"\""),
```

Comment:
I actually meant that you could just do `snippet == "println!(\"\")"`.

Reply:
I've tested that this snippet can contain whitespace or a semicolon, and your method doesn't detect them. Is there a better way I can test the presence of an empty argument?

Reply:
I don't know of any, but you're right, this should work for all realistic cases